packages/web-app-files/tests/unit/components/AppBar/SelectedResources/BatchActions.spec.js

```diff
@@ -1,134 +1,264 @@
+
+          it('should display accept and decline buttons if share status is pending', () => {
+            const store = createStore({
+              selected: addPropToSelectedFiles(selectedFiles, { status: shareStatus.pending })
+            })
+            wrapper = createShallowMountWrapper({ ...options, store })
+            actionButtons = wrapper.findAll(elSelector.ocButtonStub)
+            acceptButton = wrapper.find(elSelector.acceptButton)
+            declineButton = wrapper.find(elSelector.declineButton)
+
+            expect(actionButtons.length).toEqual(2)
+            expect(acceptButton.exists()).toBeTruthy()
+            expect(acceptButton.text()).toContain('Accept')
+            expect(declineButton.exists()).toBeTruthy()
+            expect(declineButton.text()).toContain('Decline')
+          })
+          describe('when share status is declined', () => {
+            beforeEach(() => {
+              const store = createStore({
+                selected: addPropToSelectedFiles(selectedFiles, { status: shareStatus.declined })
+              })
+              wrapper = createMountWrapper({ ...options, store })
+              actionButtons = wrapper.findAll(elSelector.ocButton)
+              acceptButton = wrapper.find(elSelector.acceptButton)
+              declineButton = wrapper.find(elSelector.declineButton)
+            })
+            it('should display accept button', () => {
+              expect(actionButtons.length).toEqual(1)
+              expect(acceptButton.exists()).toBeTruthy()
+              expect(acceptButton.text()).toContain('Accept')
+              expect(declineButton.exists()).toBeFalsy()
+            })
+            it('should call "acceptShares" when accpet button is clicked', async () => {
+              await acceptButton.trigger('click')
+
+              expect(spyAcceptShares).toHaveBeenCalledTimes(1)
+              expect(spyTriggerShareActions).toHaveBeenCalledWith(shareStatus.accepted)
+            })
+          })
+          describe('when share status is accepted', () => {
+            beforeEach(() => {
+              const store = createStore({
+                selected: addPropToSelectedFiles(selectedFiles, { status: shareStatus.accepted })
+              })
+              wrapper = createMountWrapper({ ...options, store })
+              actionButtons = wrapper.findAll(elSelector.ocButton)
+              acceptButton = wrapper.find(elSelector.acceptButton)
+              declineButton = wrapper.find(elSelector.declineButton)
+            })
+            it('should display decline button', () => {
+              expect(actionButtons.length).toEqual(1)
+              expect(declineButton.exists()).toBeTruthy()
+              expect(declineButton.text()).toContain('Decline')
+              expect(acceptButton.exists()).toBeFalsy()
+            })
+            it('should call "declineShares" when decline button is clicked', async () => {
+              await declineButton.trigger('click')
```

Comment:
it should not display accept button

Reply:
the function implementation is mocked so it's not possible to check for the changes